cypress/utils/apps/query_enhancements/inspect.js

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { QueryLanguages } from './constants';
+
+const visualizationTitlesWithNoInspectOptions = [
+  '[Flights]Controls',
+  '[Flights]MarkdownInstructions',
+  '[Flights]Delays&Cancellations',
+];
+
+const visualizationTitlesWithInspectOptions = [
+  '[Flights]AirlineCarrier',
+  '[Flights]FlightCountandAverageTicketPrice',
+  '[Flights]TotalFlights',
+  '[Flights]AverageTicketPrice',
+];
+
+/**
+ * Gets Date and returns string of date in format Jan 24, 2025 @ 16:20:08.000
+ * @param {Date} date - date to format
+ * @returns {string} - in the format e.g. Jan 24, 2025 @ 16:20:08.000
+ */
+function formatDate(date) {
+  const months = [
+    'Jan',
+    'Feb',
+    'Mar',
+    'Apr',
+    'May',
+    'Jun',
+    'Jul',
+    'Aug',
+    'Sep',
+    'Oct',
+    'Nov',
+    'Dec',
+  ];
+
+  const month = months[date.getMonth()];
+  const day = date.getDate().toString();
+  const year = date.getFullYear();
+  const hours = date.getHours().toString().padStart(2, '0');
+  const minutes = date.getMinutes().toString().padStart(2, '0');
+  const seconds = date.getSeconds().toString().padStart(2, '0');
+  const milliseconds = date.getMilliseconds().toString().padStart(3, '0');
+
+  return `${month} ${day}, ${year} @ ${hours}:${minutes}:${seconds}.${milliseconds}`;
+}
+
+/**
+ * Format value to the corresponding string expected in the UI.
+ * Formats commas in numbers, and dates.
+ * @param {string} value - field value to be formatted.
+ * @returns {string}
+ */
+const formatValue = (value) => {
+  switch (typeof value) {
+    case 'number':
+      return value.toLocaleString();
+    case 'string':
+      // SQL and PPL date string uses format YYYY-MM-DD HH:MM:SS.SSS, we want to convert to ISO date.
+      const potentialDate = new Date(value.replace(' ', 'T'));
+      if (potentialDate instanceof Date && !isNaN(potentialDate)) {
+        return formatDate(potentialDate);
+      }
+    default:
+      return value;
+  }
+};
+
+/**
+ * Returns the SavedSearchTestConfig for the provided dataset, datasetType, and language
+ * @param {string} dataset - the dataset name
+ * @param {QueryEnhancementDataset} datasetType - the type of the dataset
+ * @param {QueryEnhancementLanguageData} language - the relevant data for the query language to use
+ * @returns {InspectTestConfig}
+ */
+export const generateInspectTestConfiguration = (dataset, datasetType, language) => {
+  return {
+    dataset,
+    datasetType,
+    language: language.name,
+    testName: `dataset: ${datasetType} and language: ${language.name}`,
+  };
+};
+
+/**
+ * Flatten the object in the format e.g. {
```

Comment:
```suggestion
 /**
 * Recursively flattens a nested object structure into dot notation and formats all values
 * @param {object} obj - Object to be flattened
 * @param {string} [parentKey] - Key prefix for nested properties
 * @param {object} [properties={}] - Accumulator for flattened properties
 * @returns {object} Flattened object with formatted values
 * @example
 * const input = {
 *   FlightNum: "SYS76NQ",
 *   DestLocation: { lat: "38.94449997", lon: "-77.45580292" }
 * };
 * flattenObjectAndFormatValues(input)
 * // Returns: {
 * //   FlightNum: "SYS76NQ",
 * //   "DestLocation.lat": "38.94449997",
 * //   "DestLocation.lon": "-77.45580292"
 * // }
 */
```
better to have both input and output and don't put the example in the overall discription